### PR TITLE
GH-35847:[C++][Thirdparty] Bump xxhash version to v0.8.1

### LINKED ISSUE #35847: [C++] Bump vendored xxhash version
### Describe the enhancement requested

The latest xxhash release may have slight performance improvements.

### Component(s)

C++

### PR DESCRIPTION
**Pull request by mapleFU**
### Rationale for this change

Bump xxhash from v.0.8.0 to v0.8.1

### What changes are included in this PR?

Bump xxhash from v.0.8.0 to v0.8.1

### Are these changes tested?

Already has test

### Are there any user-facing changes?

no

* Closes: #35847

### CODE REVIEW

File: cpp/src/arrow/vendored/xxhash/README.md [hunk 1]
```diff
@@ -17,6 +17,5 @@
   under the License.
 -->
 
-The files in this directory are vendored from xxHash git tag v0.8.0
+The files in this directory are vendored from xxHash git tag v0.8.1
 (https://github.com/Cyan4973/xxHash).
-Includes https://github.com/Cyan4973/xxHash/pull/502 for Solaris compatibility
```
Comment (mapleFU) [inline cpp/src/arrow/vendored/xxhash/README.md:22]: https://github.com/Cyan4973/xxHash/commits/v0.8.1?after=35b0373c697b5f160d3db26b1cbb45a0d5ba788c+174&branch=v0.8.1&qualified_name=refs%2Ftags%2Fv0.8.1

This is already included in v0.8.1, so removed this line

**Comment by mapleFU** [comment]
@pitrou I just paste v0.8.1 release's code to our `vendor` directory. And remove a line in document

**Comment by pitrou** [comment]
@ursabot please benchmark lang=C++

**Comment by mapleFU** [comment]
test failed because ucx timeout, I think it's unrelated?

**Comment by pitrou** [comment]
Yes, it is. See https://github.com/apache/arrow/issues/35742

**Comment by pitrou** [comment]
CI failures are unrelated. Thanks for doing this @mapleFU !

**Pull request merged**